File: perl/ase_input_params_1.pl
#!/usr/bin/perl
use strict;

# Automation Scripting Essential Skills 
# Perl

# Input: Parsing command-line parameters
# Technique #1 -- Use the @ARGV array 

our $numArgs = $#ARGV;

if( $numArgs >= 0 )
{
    print ("There were " . ($numArgs + 1) . " args passed.\n");
    for (my $ndx = 0 ; $ndx <= $numArgs ; $ndx += 1 )
    {
        print "$ARGV[$ndx]\n";
    }
}

File: python3/ase_input_params_1.py
#!/usr/bin/env python3

# Automation Scripting Essential Skills 
# Python3

# Input: Parsing command-line parameters 
# Technique1 -- Use sys.argv array 
import sys

print(f"You passed {len(sys.argv) -1} parameters.")

# since I want to list the arg #'s too
for ndx in range(1, len(sys.argv)):
    print(f"param# {ndx}: {sys.argv[ndx]}")
